.github/workflows/CD.yml

```
@@ -495,72 +486,73 @@ jobs:
+      - name: Cache if success
+        id: release-latest-image
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
-          path: build/packages
-
-      - name: Load Image
-        run: docker load < build/packages/$IMAGE_NAME.tar
+          path: |
+            VERSION
+          key: release-latest-image-${{ github.run_id }}
 
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Restore cached packages
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        uses: actions/cache@v2
         with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2
+          key: "${{ env.PACKAGE_CACHE_KEY }}"
+          path: $PACKAGING_ROOT
 
       - name: Compare version with Dockerhub latest
         id: version
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         run: |
-          TAG=`cat build/packages/VERSION`
+          TAG="${{ needs.release-checking.outputs.version }}"
           TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
 
-      - name: Tag latest and upload to ecr
-        if: steps.version.outputs.any-update == 'true'
-        run: |
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/aws-observability
-          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:latest
-          docker push public.ecr.aws/$ECR_REPO:latest
-
       - name: Login Dockerhub
         uses: docker/login-action@v1
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         with:
           username: "${{ secrets.DOCKERHUB_RELEASE_USERNAME }}"
           password: "${{ secrets.DOCKERHUB_RELEASE_TOKEN }}"
 
-      - name: Tag latest and upload to docker
-        if: steps.version.outputs.any-update == 'true'
-        run: |
-          REPO_NAME="$IMAGE_NAMESPACE/$IMAGE_NAME"
-          docker tag $IMAGE_NAME $REPO_NAME:latest
-          docker push $REPO_NAME:latest
+      - name: Login to Public Integration Test ECR
+        id: login-ecr
+        if: ${{ steps.release-latest-image.outputs.cache-hit != 'true' && steps.version.outputs.any-update == 'true' }}
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.RELEASE_KEY_ID }}
+          password: ${{ secrets.RELEASE_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
+      - name: Pull image from integration test ECR, tag as latest and push to public release ECR and DockerHub
+        if: ${{ steps.release-latest-image.outputs.cache-hit != 'true'  && steps.version.outputs.any-update == 'true' }}
+        uses: akhilerm/tag-push-action@v2.0.0
+        with:
+          src: public.ecr.aws/$ECR_REPO_INTEGRATION_TEST:${{ needs.release-checking.outputs.testing_version }}
```

```suggestion
          src: public.ecr.aws/$ECR_REPO:${{ needs.release-checking.outputs.version }}
```
Still think this should be the one already pushed into the release ECR. They're technically the same image, so it doesn't really matter.

---

True thought. Fixed it in the newest commit though.